File: programmers/Level2/카펫.py
#완전탐색
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    
    for i in range(2, total-2):
        if total % i == 0:
            for j in range(1, i+1):
                cx = (2*i)+(2*(j-2))
                cy = (i-2)*(j-2)
                if brown == cx and yellow == cy:
                    answer.append(i)
                    answer.append(j)
                    break
    
    return answer

File: JAVA_programmers/Level 2/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int total = brown+yellow;
        
        for(int r=3; r<=total; r++){
            if(total%r != 0) continue;
            int c = total/r;
            
            int cb = (2*r)+(2*(c-2));
            int cy = (r-2)*(c-2);
            if(cb==brown && cy==yellow){
                answer[0] = r;
                answer[1] = c;
            }
        }        
        return answer;
    }
}
